pydantic/_internal/_generate_schema.py

```diff
@@ -117,6 +124,26 @@ def generate_schema(self, obj: Any) -> core_schema.CoreSchema:  # noqa: C901
             if issubclass(obj, dict):
                 return self._dict_subclass_schema(obj)
             # probably need to take care of other subclasses here
+        elif isinstance(obj, typing.TypeVar):
+            # TODO: Add a check for recursion depth exception and an explanatory re-raise
+            #   Idea: you shouldn't create a cycle of typevar substitutions. I don't think this is ever sensible.
+            if self.typevars_map is not None and obj in self.typevars_map:
+                mapped = self.typevars_map[obj]
+                if mapped is not obj:
+                    try:
+                        return self.generate_schema(mapped)
+                    except RecursionError as exc:
+                        raise TypeError(
+                            f'The maximum recursion depth was exceeded while generating the schema for a TypeVar. '
+                            f'This likely indicates a cycle in the TypeVar substitutions map ({self.typevars_map=}). '
+                            'This may be resolved by using the TypeVars in the same order as the original '
+                            'parameterization, or by using entirely new ones.'
+                        ) from exc
+            if obj.__bound__:
+                return self.generate_schema(obj.__bound__)
+            if obj.__constraints__:
+                return self._union_schema(typing.Union[obj.__constraints__])
+            return core_schema.AnySchema(type='any')
```

Comment:
I did think about this; I spent some time thinking through the implications of disallowing model classes with free typevars from being instantiated at all, and ultimately I think this approach is preferable.

I couldn't think of any cases where an error was merited that mypy wouldn't be able to catch with an appropriate configuration if you wanted to be aggressive about your types (i.e., using generic models with implicit Any parameter). And it felt to me that the flexibility benefits of being able to work with unspecified generics was worth more than other forms of strictness here.

-----

The only pattern that I could imagine being problematic would be something along the following lines:
```python
class MyGenericModel(BaseModel, Generic[T]):
    my_field: T

MyConcreteModel = MyGenericModel[MyType]
```
If there was a non-pathological example where `MyGenericModel(my_field=instance_of_my_type).my_field` could end up different than `MyConcreteModel(my_field=instance_of_my_type).my_field` due to (again, non-pathological) validator usage, etc., I think that would be concerning. I couldn't think of anything that would cause this though without going out of our way (e.g., by having non-idempotent validators or something like that).

Reply:
okay, let's leave it.